### PR TITLE
Fix examining of items with a capitalized access key when trading

### LINKED ISSUE #23521: NPC Trade UI: '?' inspect does not work for items with capitalized access keys
Game version: 0.C-26948-gbbcf229

### Expected behavior

Inspect via `?` in the trade UI for NPC should work for items outside the [a-z] shortcut accessors.

### Actual behavior

Inspect via `?` in the trade UI for NPC does not work if the item has an access key that is outside [a-z].

### Steps to reproduce the behavior
Have over 26 trade-able items, talk to NPC, select trade, select `?`, try to inspect/examine item with capitalized access key.

### PR DESCRIPTION
**Pull request by iordanivanov**
#### Summary
```SUMMARY: Bugfixes "Fix examining of items with a capitalized access key when trading"```

#### Purpose of change
Fixes #23521.

#### Describe the solution
Added an if / else if / else block when collecting the input event in order to subtract the correct ASCII decimal code, as well as add 26 when a capitalized access key is selected (so if 'A' is selected, the 27th item (counting from one) in the list will be examined).

### CODE REVIEW

**Comment by iordanivanov** [comment]
I opened the pull request before noticing that my indentation does not line up with the rest of the code. I'm sorry for this, how should I fix it? Is it ok if I do it in a separate commit?

**Comment by anothersimulacrum** [comment]
Yeah, it's OK if you do it in a separate commit, and there are a few ways to only do it with one commit if you want to.

**Comment by iordanivanov** [comment]
I just saw that a bit further down in the same file, there's the exact same block that I added in, albeit slightly different:
```cpp
default: // Letters & such
                if( ch >= 'a' && ch <= 'z' ) {
                    ch -= 'a';
                } else if( ch >= 'A' && ch <= 'Z' ) {
                    ch = ch - 'A' + ( 'z' - 'a' ) + 1;
                } else {
                    continue;
                }
```
It doesn't feel right to have two snippets of code that do the same thing written differently (they're also quite close as well). Should I edit mine to look like the already existing one? And, to avoid duplicating code, is there a function somewhere that does this already?

File: src/npctrade.cpp [hunk 1]
```diff
@@ -328,7 +328,17 @@ TAB key to switch lists, letters to pick items, Enter to finalize, Esc to quit,\
                 draw_border( w_tmp );
                 wrefresh( w_tmp );
                 // TODO: use input context
-                help = inp_mngr.get_input_event().get_first_input() - 'a';
+                help = inp_mngr.get_input_event().get_first_input();
+                if( help >= 'a' && help <= 'z' ) {
+                    help -= 'a';
+                }
+                else if( help >= 'A' && help <= 'Z' ) {
```
Comment (Night-Pryanik) [inline src/npctrade.cpp]: Move `else if` clause one line up, so it should look like `} else if(...`. The same with `else` below.

File: src/npctrade.cpp [hunk 2]
```diff
@@ -328,7 +328,17 @@ TAB key to switch lists, letters to pick items, Enter to finalize, Esc to quit,\
...
+                    help = ( help - 'A' ) + 26;
```
Comment (Night-Pryanik) [inline src/npctrade.cpp]: Brackets are redundant.

**Comment by mlangsdorf** [comment]
If you have two pieces of code that do the same thing in the same file, normally I would recommend extracting them both to a common function.  In this case, you want to generate the `continue` under some circumstances so that may not be feasible here.

**Pull request merged**